### DISCUSSION TITLE
Can't import pycuda.driver: undefined symbol: cuIpcOpenMemHandle_v2

### DISCUSSION THREAD

**Discussion by Aldenhovel** [inducer/pycuda#409] [Q&A] [answered]
Hi, I can't import pycuda.driver while i am trying to run NVIDIA/TensorRT's Python sample "introductory_parser_samples".
When I try:
```
import pycuda.driver as cuda
```
then report error:
```
 UserWarning: Failed to import the CUDA driver interface, with an error message indicating that the version of your CUDA header does not match the version of your CUDA driver.
  warn(
Traceback (most recent call last):
  File "/home/xavier/trainee/v723/TensorRT/samples/python/introductory_parser_samples/onnx_resnet50.py", line 33, in <module>
    import pycuda.driver as cuda
  File "/home/xavier/anaconda3/envs/trt723/lib/python3.9/site-packages/pycuda/driver.py", line 66, in <module>
    from pycuda._driver import *  # noqa
ImportError: /home/xavier/anaconda3/envs/trt723/lib/python3.9/site-packages/pycuda/_driver.cpython-39-aarch64-linux-gnu.so: undefined symbol: cuIpcOpenMemHandle_v2
```
Which version of pycuda should I install? Thanks.

I am using Nvidia Jetson AGX Xavier mechine, this is the environment:
- Python 3.9.16
- Ubuntu 18.04
- CUDA 10.2.89
- pycuda 2022.2.2

**Comment by inducer** [comment]
Did you build PyCUDA from source? If not, where did you get the binary?

**Reply by Aldenhovel** [reply]
I got it by `pip install pycuda` ,should I build it from source?

**Reply by inducer** [reply, marked as answer]
You must've already built it from source then. In this case, your CUDA headers must not match the `libcuda.so.1` on your system (or you might have multiple of either, and it's using the wrong version). You may have multiple installations of CUDA floating around. Try to delete all but one and rebuild PyCUDA.
